File: common/src/main/java/com/ssblur/scriptor/word/descriptor/power/OverwhelmingStrengthDescriptor.java
package com.ssblur.scriptor.word.descriptor.power;

import com.ssblur.scriptor.effect.ScriptorEffects;
import com.ssblur.scriptor.helpers.targetable.EntityTargetable;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.word.descriptor.CastDescriptor;
import com.ssblur.scriptor.word.descriptor.Descriptor;
import net.minecraft.world.effect.MobEffectInstance;
import net.minecraft.world.entity.LivingEntity;

public class OverwhelmingStrengthDescriptor extends Descriptor implements CastDescriptor, StrengthDescriptor {
  static Cost COST = new Cost(0, COSTTYPE.ADDITIVE);

  public OverwhelmingStrengthDescriptor() {}

  @Override
  public Cost cost() {
    return COST;
  }

  @Override
  public double strengthModifier() {
    return 20;
  }

  @Override
  public boolean allowsDuplicates() {
    return false;
  }

  @Override
  public boolean cannotCast(Targetable caster) {
    if(caster instanceof EntityTargetable entityTargetable && entityTargetable.getTargetEntity() instanceof LivingEntity living) {
      living.addEffect(new MobEffectInstance(ScriptorEffects.MUTE.get(), 20 * 60));
      return false;
    }
    return true;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/power/OverwhelmingStrengthDescriptor.kt
package com.ssblur.scriptor.word.descriptor.power

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.effect.ScriptorEffects.MUTE
import com.ssblur.scriptor.helpers.targetable.EntityTargetable
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.descriptor.CastDescriptor
import net.minecraft.world.effect.MobEffectInstance
import net.minecraft.world.entity.LivingEntity

class OverwhelmingStrengthDescriptor : Descriptor(), CastDescriptor, StrengthDescriptor {
    override fun cost(): Cost {
        return COST
    }

    override fun strengthModifier(): Double {
        return 20.0
    }

    override fun cannotCast(caster: Targetable?): Boolean {
        if (caster is EntityTargetable && caster.targetEntity is LivingEntity) {
            val living = caster.targetEntity as LivingEntity
            living.addEffect(MobEffectInstance(MUTE.ref(), 20 * 60))
            return false
        }
        return true
    }

    companion object {
        var COST: Cost = Cost(0.0, COSTTYPE.ADDITIVE)
    }
}
